connector/exceptionsconnector/connector_logs.go

```diff
@@ -0,0 +1,132 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exceptionsconnector
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	conventions "go.opentelemetry.io/collector/semconv/v1.6.1"
+	"go.uber.org/zap"
+)
+
+type logsConnector struct {
+	logsConsumer consumer.Logs
+	component.StartFunc
+	component.ShutdownFunc
+
+	logger *zap.Logger
+	ld     plog.Logs
+}
+
+func newLogsConnector(logger *zap.Logger) (*logsConnector, error) {
+	logger.Info("Building logs exceptionsconnector")
+
+	return &logsConnector{
+		logger: logger,
+		ld:     plog.NewLogs(),
+	}, nil
+}
+
+// Capabilities implements the consumer interface.
+func (c *logsConnector) Capabilities() consumer.Capabilities {
+	return consumer.Capabilities{MutatesData: false}
+}
+
+// ConsumeTraces implements the consumer.Traces interface.
+// It aggregates the trace data to generate logs.
+func (c *logsConnector) ConsumeTraces(ctx context.Context, traces ptrace.Traces) error {
+	c.logger.Debug("Consume traces")
+	for i := 0; i < traces.ResourceSpans().Len(); i++ {
+		rspans := traces.ResourceSpans().At(i)
+		resourceAttr := rspans.Resource().Attributes()
+		serviceAttr, ok := resourceAttr.Get(conventions.AttributeServiceName)
+		if !ok {
+			continue
+		}
+		serviceName := serviceAttr.Str()
+		ilsSlice := rspans.ScopeSpans()
```

Comment:
Should we preserve the scope information? i.e. group the logs in the same way that the spans were grouped? We do this in the count connector for [example](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/7c0fd3de2aa7cbf61d6360c6208079c761992512/connector/countconnector/connector.go#L79).

Reply:
Done, let me know if it's correct

Reply:
Sorry, I misunderstood your comment. You are right, I was creating everytime a new scope. Does [this](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/21090/commits/335665bbde11ca13de9d568f0864bdbe47e10ae7) make more sense?

Reply:
It's not quite what I was suggesting but I don't want to hold up the component for what might be a non-issue to many users. We can revisit based on feedback.

Reply:
I agree with @djaglowski here, I would expect the same resource/scope from the span to be converted as part of the new log entry for the exception.

Reply:
So if I understand correctly, `sl := c.newScopeLogs()` should be inside of the first for loop?

Reply:
It should be in the loop starting in line 74, and `sl := c.newScopeLogs()` should have the scope name from the `ils`, instead of a new scope name. Something like:

```suggestion
		ilsSlice := rspans.ScopeSpans()
		for j := 0; j < ilsSlice.Len(); j++ {
			ils := ilsSlice.At(j)
			sl := plog.NewScopeLogs()
			ils.Scope().CopyTo(sl.Scope())
```